api/provider/aws/service_read.go

```diff
@@ -55,12 +74,91 @@ func (s *ServiceProvider) readService(clusterName, serviceID string) (*ecs.Servi
 
 	output, err := s.AWS.ECS.DescribeServices(input)
 	if err != nil {
+		if err, ok := err.(awserr.Error); ok && err.Code() == "ServiceNotFoundException" {
+			return nil, errors.Newf(errors.ServiceDoesNotExist, "Service '%s' does not exist", serviceID)
+		}
+
+		return nil, err
+	}
+
+	if len(output.Services) == 0 {
+		return nil, errors.Newf(errors.ServiceDoesNotExist, "Service '%s' does not exist", serviceID)
+	}
+
+	return output.Services[0], nil
+}
+
+func (s *ServiceProvider) lookupDeployIDFromTaskDefinitionARN(taskDefinitionARN string) (string, error) {
+	tags, err := s.TagStore.SelectByType("deploy")
+	if err != nil {
+		return "", err
+	}
+
+	if tag, ok := tags.WithKey("arn").WithValue(taskDefinitionARN).First(); ok {
+		return tag.EntityID, nil
+	}
+
+	return "", errors.Newf(errors.DeployDoesNotExist, "Failed to find deploy with ARN '%s'", taskDefinitionARN)
+}
+
+func (s *ServiceProvider) makeDeploymentModel(deployID string) (*models.Deployment, error) {
+	model := &models.Deployment{
+		DeployID: deployID,
+	}
+
+	tags, err := s.TagStore.SelectByTypeAndID("deploy", deployID)
+	if err != nil {
 		return nil, err
 	}
 
-	for _, service := range output.Services {
-		return service, nil
+	if tag, ok := tags.WithKey("name").First(); ok {
+		model.DeployName = tag.Value
+	}
+
+	if tag, ok := tags.WithKey("version").First(); ok {
+		model.DeployVersion = tag.Value
+	}
+
+	return model, nil
+}
+
+func (s *ServiceProvider) makeServiceModel(environmentID, loadBalancerID, serviceID string) (*models.Service, error) {
+	model := &models.Service{
+		EnvironmentID:  environmentID,
+		LoadBalancerID: loadBalancerID,
+		ServiceID:      serviceID,
+	}
+
+	tags, err := s.TagStore.SelectByTypeAndID("service", serviceID)
+	if err != nil {
+		return nil, err
 	}
 
-	return nil, fmt.Errorf("ecs service '%s' in cluster '%s' does not exist", serviceID, clusterName)
+	if tag, ok := tags.WithKey("name").First(); ok {
+		model.ServiceName = tag.Value
+	}
+
+	if environmentID != "" {
```

Comment:
I don't think this block is required. It should be safe to assume `environmentID` will never be empty string (unlike loadBalancerID)

Reply:
✅